cfg/1.1.0/definitions.yaml

```diff
@@ -4313,10 +4313,13 @@ groups:
             
     - id: 3.2.1.c
       description: "Ensure source routed packets are not accepted"
-      audit: "grep net.ipv4.conf.all.accept_source_route /etc/sysctl.conf /etc/sysctl.d/*"   
+      audit: "grep ^[^#]net.ipv4.conf.all.accept_source_route /etc/sysctl.conf /etc/sysctl.d/*"   
```

Comment:
I don't get it? The command in the CIS spec is 

```
grep "net\.ipv4\.conf\.all\.accept_source_route" /etc/sysctl.conf /etc/sysctl.d/*
```

(Note that the # in the CIS spec is supposed to indicate the prompt, not a comment line.)

Reply:
Yes but we get false positive for stuff that comes built-in as a comment, like in this case the file /etc/sysctl.conf contains the comment:

> #net\.ipv4\.conf\.all\.accept_source_route = 0

which will pass the current test, so we just need to filter it out, like with " **^net...** ", and we know for sure it's not a comment but the real intended line we want to test

Reply:
I don't think this is quite right, at the moment this is looking for exactly one character that isn't # between the beginning of the line, and `net...`

```
vagrant@kube:~$ echo "#net" | grep "^[^#]net"
vagrant@kube:~$ echo "net" | grep "^[^#]net"
vagrant@kube:~$ echo "anet" | grep "^[^#]net"
anet
vagrant@kube:~$ echo " net" | grep "^[^#]net"
 net
vagrant@kube:~$ echo "  net" | grep "^[^#]net"
```